plugins/httpapi/nd.py

```diff
@@ -223,11 +223,12 @@ def send_request(self, method, path, data=None):
             raise ConnectionError(json.dumps(self._verify_response(None, method, full_path, None)))
         return self._verify_response(response, method, full_path, rdata)
 
-    def send_file_request(self, method, path, file=None, data=None, remote_path=None, file_key="file"):
+    def send_file_request(self, method, path, file=None, data=None, remote_path=None, file_key="file", file_ext=None):
         """This method handles file download and upload operations
         :arg method (str): Method can be GET or POST
         :arg path (str): Path should be the resource path
         :arg file (str): The absolute file path of the target file
+        :arg file_ext (str): The file extension, with leading dot, to be used for the file. If file has already an extension, it will be replaced
```

Comment:
should we add a check that the file extension leads with a dot? possibly error or inject when this is not done.

Reply:
Sounds good, added